### PR TITLE
fix: support `alter table <tableName> drop index <indexName>`  see issueNo: 16254

### LINKED ISSUE #16254: not support `alter table <tableName> drop index <indexName>` grammar
### Which version of ShardingSphere did you use?

`5.1.1`

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

`ShardingSphere-Proxy`

### Expected behavior

shardingRule is: 

```yaml

schemaName: sharding_db

dataSources:
  ds_0:
    url: jdbc:mysql://127.0.0.1:3306/demo_ds_0?serverTimezone=UTC&useSSL=false
    username: root
    password: root
    connectionTimeoutMilliseconds: 30000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 50
    minPoolSize: 1
  ds_1:
    url: jdbc:mysql://127.0.0.1:3306/demo_ds_1?serverTimezone=UTC&useSSL=false
    username: root
    password: root
    connectionTimeoutMilliseconds: 30000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 50
    minPoolSize: 1

rules:
- !SHARDING
  tables:
    t_order:
      actualDataNodes: ds_${0..1}.t_order_${0..1}
      tableStrategy:
        standard:
          shardingColumn: order_id
          shardingAlgorithmName: t_order_inline
      keyGenerateStrategy:
        column: order_id
        keyGeneratorName: snowflake
    t_order_item:
      actualDataNodes: ds_${0..1}.t_order_item_${0..1}
      tableStrategy:
        standard:
          shardingColumn: order_id
          shardingAlgorithmName: t_order_item_inline
      keyGenerateStrategy:
        column: order_item_id
        keyGeneratorName: snowflake
  bindingTables:
    - t_order,t_order_item
  defaultDatabaseStrategy:
    standard:
      shardingColumn: user_id
      shardingAlgorithmName: database_inline
  defaultTableStrategy:
    none:

  shardingAlgorithms:
    database_inline:
      type: INLINE
      props:
        algorithm-expression: ds_${user_id % 2}
    t_order_inline:
      type: INLINE
      props:
        algorithm-expression: t_order_${order_id % 2}
    t_order_item_inline:
      type: INLINE
      props:
        algorithm-expression: t_order_item_${order_id % 2}

  keyGenerators:
    snowflake:
      type: SNOWFLAKE
```

alter table <tableName> drop index <indexName> return success

### Actual behavior

image

### Reason analyze (If you can)

this is a bug.

### PR DESCRIPTION
**Pull request by ICannerxy**
Fixes #16254.

Changes proposed in this pull request:
- support `alter table <tableName> drop index <indexName>` grammar

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/impl/MySQLDDLStatementSQLVisitor.java [hunk 1]
```diff
@@ -333,6 +333,11 @@ public ASTNode visitAlterList(final AlterListContext ctx) {
                     result.getValue().add(new DropConstraintDefinitionSegment(ctx.getStart().getStartIndex(), ctx.getStop().getStopIndex(), constraintSegment));
                 } else if (null == alterTableDrop.KEY() && null == alterTableDrop.keyOrIndex()) {
                     result.getValue().add(generateDropColumnDefinitionSegment(alterTableDrop));
+                } else if (null != alterTableDrop.keyOrIndex()) {
+                    IndexSegment indexSegment = (IndexSegment) visit(alterTableDrop.indexName());
+                    ConstraintSegment constraintSegment = new ConstraintSegment(alterTableDrop.indexName().getStart().getStartIndex(), alterTableDrop.indexName().getStop().getStopIndex(),
+                            indexSegment.getIdentifier());
+                    result.getValue().add(new DropConstraintDefinitionSegment(ctx.getStart().getStartIndex(), ctx.getStop().getStopIndex(), constraintSegment));
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/impl/MySQLDDLStatementSQLVisitor.java:340]: @ICannerxy Why parse drop index segment into DropConstraintDefinitionSegment?

**Review by strongduanmu — CHANGES_REQUESTED** [review]
@ICannerxy Thank you for your pr, can you add some unit test for this logic? Besides, please modify the pr title for friendly search.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/segment/ddl/index/DropIndexDefinitionSegment.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+package org.apache.shardingsphere.sql.parser.sql.common.segment.ddl.index;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.apache.shardingsphere.sql.parser.sql.common.segment.ddl.AlterDefinitionSegment;
+
+@RequiredArgsConstructor
+@Getter
+public class DropIndexDefinitionSegment implements AlterDefinitionSegment {
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/segment/ddl/index/DropIndexDefinitionSegment.java:26]: @ICannerxy Please add final for DropIndexDefinitionSegment, and add java doc.

**Review by strongduanmu — CHANGES_REQUESTED** [review]
@ICannerxy Please add sql parse test case for drop index parse.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/segment/ddl/index/DropIndexDefinitionSegment.java (hunk 2 above)
Comment (ICannerxy) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/segment/ddl/index/DropIndexDefinitionSegment.java]: done

File: shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/ddl/AlterTableStatementContextTest.java [hunk 3]
```diff
@@ -110,12 +112,16 @@ private void assertNewInstance(final AlterTableStatement alterTableStatement) {
         when(dropConstraintDefinition.getConstraintName()).thenReturn(constraint);
         when(alterTableStatement.getDropConstraintDefinitions()).thenReturn(Collections.singletonList(dropConstraintDefinition));
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/statement/ddl/AlterTableStatementContextTest.java:114]: Please remove these useless blank lines.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Review by strongduanmu — APPROVED** [review]
@ICannerxy Thank you for your great contribution, merged.

**Pull request merged**